File: src/main/java/fr/guddy/roombookings/domain/room/SimpleRoom.java
package fr.guddy.roombookings.domain.room;

import java.util.Map;
import org.cactoos.map.MapEntry;
import org.cactoos.map.MapOf;

public record SimpleRoom(String name, int capacity) implements Room {
  @Override
  public Map<String, Object> map() {
    return new MapOf<String, Object>(
      new MapEntry<>("name", this.name),
      new MapEntry<>("capacity", this.capacity)
    );
  }
}

File: src/main/java/fr/guddy/roombookings/domain/room/SimpleRoom.kt
package fr.guddy.roombookings.domain.room

class SimpleRoom(private val name: String, private val capacity: Int) : Room {

    override fun name(): String {
        return name
    }

    override fun capacity(): Int {
        return capacity
    }

    override fun map(): Map<String, Any> = linkedMapOf(
            "name" to name,
            "capacity" to capacity
    )
}
